itest/echo_test.go

```diff
@@ -0,0 +1,86 @@
+package itest
+
+import (
+	"context"
+	"testing"
+
+	"github.com/libp2p/go-libp2p"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/peerstore"
+)
+
+func createEchos(t *testing.T, count int, opts ...libp2p.Option) []*Echo {
+	result := make([]*Echo, 0, count)
+
+	for i := 0; i < count; i++ {
+		h, err := libp2p.New(opts...)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		e := NewEcho(h)
+		result = append(result, e)
+	}
+
+	for i := 0; i < count; i++ {
+		for j := 0; j < count; j++ {
+			if i == j {
+				continue
+			}
+
+			result[i].Host.Peerstore().AddAddrs(result[j].Host.ID(), result[j].Host.Addrs(), peerstore.PermanentAddrTTL)
+		}
+	}
+
+	return result
+}
+
+func checkEchoStatus(t *testing.T, e *Echo, expected EchoStatus) {
```

Comment:
This could be as simple as `require.Equal(t, e.Status(), expected)`, or am I missing something here?

Reply:
yes, assuming we use `require` and friends; easy change.

Reply:
done.